packages/core-components/src/components/Avatar/Avatar.tsx

```diff
@@ -67,15 +67,23 @@ export interface AvatarProps {
 export function Avatar(props: AvatarProps) {
   const { displayName, picture, customStyles } = props;
   const classes = useStyles();
+  let styles = { ...customStyles };
+  // We only calculate the background color if there's not an avatar
+  // picture. If there is a picture, it might bave a transparent
+  // background and we don't know whether the calculated background
+  // color will clash.
+  if (!picture) {
```

Comment:
Would you mind adding a test for this in `Avatar.test.tsx`?

Reply:
Added a test. For the second case, I am not sure how to test that the `background-color` style is omitted altogether, so it's overconstrained to test that only the generated value is omitted.